tests/ui/redundant_guards_const_fn.rs

```diff
@@ -0,0 +1,47 @@
+#![allow(clippy::single_match)]
```

Comment:
Is there a reason that this is its own test file, and not just part of the existing `redundant_guards.rs` test? Typically we have a single test file for lints, and optionally a `mod issue0000 {}`, [like here](https://github.com/rust-lang/rust-clippy/blob/a2c1d565e576efb5f3abce367723711afe65c8cd/tests/ui/redundant_guards.rs#L205).

Reply:
I created mod issue12243 in `redundant_guards.rs` and moved the test cases there.